src/mapping/toroidal_to_cylindrical.hpp

```diff
@@ -0,0 +1,442 @@
+    using Z_cov = typename Z::Dual;
+    /// @brief The covariant form of the third physical coordinate.
+    using Zeta_cov = typename Zeta::Dual;
+    /// @brief The covariant form of the first logical coordinate.
+    using Rho_cov = typename Rho::Dual;
+    /// @brief The covariant form of the second logical coordinate.
+    using Theta_cov = typename Theta::Dual;
+    /// @brief The covariant form of the third logical coordinate.
+    using Phi_cov = typename Phi::Dual;
+
+private:
+    using spline_idx_range = IdxRange<BSplineRho, BSplineTheta>;
+
+    using SplineType = DConstField<spline_idx_range, MemorySpace>;
+
+    using IdxRangeRhoTheta = typename SplineEvaluator::evaluation_domain_type;
+    using IdxRangeTheta = typename SplineEvaluator::evaluation_domain_type2;
+    using IdxTheta = typename IdxRangeTheta::discrete_element_type;
+
+private:
+    SplineType m_R_spline_representation;
+    SplineType m_Z_spline_representation;
+    SplineEvaluator m_spline_evaluator;
+    IdxRangeRTheta m_idx_range_singular_point;
+
+public:
+    /**
+     * @brief Instantiate a DiscreteToCartesian from the coefficients of 2D splines approximating the mapping.
+     *
+     * A discrete mapping is a mapping whose values are known only at the mesh points of the grid.
+     * To interpolate the mapping, we use B-splines. The DiscreteToCartesian mapping is initialised
+     * from the coefficients in front of the basis splines which arise when we approximate the
+     * functions @f$ R(\rho,\theta) @f$, and @f$ Z(\rho,\theta) @f$ (with @f$ R @f$ and @f$ Z @f$ the physical dimensions in
+     * the logical domain) with Splines (using SplineBuilder2D). Then to interpolate the mapping,
+     * we will evaluate the decomposed functions on B-splines (see DiscreteToCartesian::operator()).
+
+     *
+     * Here, the evaluator is given as input.
+     *
+     * @param[in] curvilinear_to_R
+     * 		Bsplines coefficients of the first physical dimension in the logical domain.
+     *
+     * @param[in] curvilinear_to_Z
+     * 		Bsplines coefficients of the second physical dimension in the logical domain.
+     *
+     * @param[in] evaluator
+     * 		The evaluator used to evaluate the mapping.
+     *
+     * @param[in] idx_range_singular_point
+     *      The index range describing the points which should be used to evaluate functions at the central point.
+     *
+     *
+     * @see SplineBuilder2D
+     * @see DiscreteToCartesian::operator()
+     * @see SplineBoundaryValue
+     */
+    KOKKOS_FUNCTION DiscreteToCartesian(
+            SplineType curvilinear_to_R,
+            SplineType curvilinear_to_Z,
+            SplineEvaluator const& evaluator,
+            IdxRangeRTheta idx_range_singular_point)
+        : m_R_spline_representation(curvilinear_to_R)
+        , m_Z_spline_representation(curvilinear_to_Z)
+        , m_spline_evaluator(evaluator)
+        , m_idx_range_singular_point(idx_range_singular_point)
+    {
+    }
```

Comment:
Apart from an orthogonal direction, what does this class add on top of `DiscreteToCartesian`? Is it not possible to create a:
```cpp
template <
        class DiscreteToCartesianMapping,
        class Zeta>
class ToroidalToCylindrical;
```
to avoid duplicating the logic in `DiscreteToCartesian`?

Reply:
Or to extend `DiscreteToCartesian`?
```
template <
        class R,
        class Z,
        class Zeta,
        class SplineEvaluator,
        class Rho = typename SplineEvaluator::continuous_dimension_type1,
        class Theta = typename SplineEvaluator::continuous_dimension_type2,
        class MemorySpace = typename SplineEvaluator::memory_space>
class ToroidalToCylindrical : public DiscreteToCartesian<R, Z, SplineEvaluator, Rho, Theta, MemorySpace>
```